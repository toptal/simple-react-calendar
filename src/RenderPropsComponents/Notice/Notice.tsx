import React, { FC } from 'react'

type Props = {
  blockClassName?: string
}

const displayName = 'Notice'

const Notice: FC<Props> = ({ blockClassName, children }) => {
  if (!children) return null

  return <div className={`${blockClassName}-notice`}>{children}</div>
}

Notice.displayName = displayName

export default Notice
