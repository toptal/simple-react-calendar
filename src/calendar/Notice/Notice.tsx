import React, { FC } from 'react'

import { INoticeType } from '../../@types'

export interface Props {
  blockClassName: string
  type: INoticeType
}

const Notice: FC<Props> = ({type, blockClassName}) => {
  const content =
    type === 'overlapping_with_disabled' ||
    (type === 'disabled_day_click' && 'Selected range must not overlap with disabled dates.')

  return <div className={`${blockClassName}-notice`}>{content}</div>
}

Notice.displayName = 'Notice'

export default Notice
