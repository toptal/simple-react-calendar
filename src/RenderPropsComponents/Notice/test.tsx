import { render } from '@testing-library/react'
import React, { ComponentProps } from 'react'

import Notice from './Notice'

const renderNotice = (props: Partial<ComponentProps<typeof Notice>> = {}) =>
  render(<Notice {...props} />)

test('default render', () => {
  const { container } = renderNotice({ children: 'Example notice error', blockClassName: 'example-class-name' })

  expect(container).toMatchSnapshot()
})

describe('when no `children` defined', () => {
  test('renders null', () => {
    const { container } = renderNotice({ })

    expect(container).toMatchSnapshot()
  })
})
