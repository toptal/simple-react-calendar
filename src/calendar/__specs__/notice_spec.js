import React from 'react'

import ReactTestRenderer from 'react-test-renderer'
import Notice from '../notice'

describe('Notice', () => {
  let tree

  context('when prop `type` is "overlapping_with_disabled"', () => {
    it('renders <Notice />', () => {
      tree = ReactTestRenderer.create(<Notice type="overlapping_with_disabled" />).toJSON()

      expect(tree).toMatchSnapshot()
    })

    context('when prop `blockClassName` is "Whatever"', () => {
      it('renders div with a custom block class name', () => {
        tree = ReactTestRenderer.create(<Notice blockClassName="Whatever" type="overlapping_with_disabled" />).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })

  context('when prop `type` is "disabled_day_click"', () => {
    it('renders <Notice />', () => {
      tree = ReactTestRenderer.create(<Notice type="disabled_day_click" />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
