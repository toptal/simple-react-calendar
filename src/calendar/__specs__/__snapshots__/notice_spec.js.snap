// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notice when prop \`type\` is "disabled_day_click" renders <Notice /> 1`] = `
<div
  className="calendar-notice"
>
  Selected range must not overlap with disabled dates.
</div>
`

exports[`Notice when prop \`type\` is "overlapping_with_disabled" renders <Notice /> 1`] = `
<div
  className="calendar-notice"
>
  Selected range must not overlap with disabled dates.
</div>
`

exports[
  `Notice when prop \`type\` is "overlapping_with_disabled" when prop \`blockClassName\` is "Whatever" renders div with a custom block class name 1`
] = `
<div
  className="Whatever-notice"
>
  Selected range must not overlap with disabled dates.
</div>
`
